Cleaning Log I

In [1]:
import pandas as pd
import seaborn as sns

df = pd.read_csv("allbacks.csv")
df

,rownames,volume,area,weight,cover
0,1,885,382,800,hb
1,2,1016,468,950,hb
2,3,1125,387,1050,hb
3,4,239,371,350,hb
4,5,701,371,750,hb
5,6,641,367,600,hb
6,7,1228,396,1075,hb
7,8,412,0,250,pb
8,9,953,0,700,pb
9,10,929,0,650,pb


Cleaning Log II

In [2]:
coverSort = df.sort_values("cover", axis=0, ascending=True, na_position='last')
coverSort

,rownames,volume,area,weight,cover
0,1,885,382,800,hb
1,2,1016,468,950,hb
2,3,1125,387,1050,hb
3,4,239,371,350,hb
4,5,701,371,750,hb
5,6,641,367,600,hb
6,7,1228,396,1075,hb
7,8,412,0,250,pb
8,9,953,0,700,pb
9,10,929,0,650,pb


Tne median of 'cover' is between 7 and 8.

Cleaning Log III

In [3]:
print(f'The mean weight of all books is {df['weight'].mean()} grams.')

The mean weight of all books is 706.6666666666666 grams.


Cleaning Log IV

In [4]:
hbBooks = coverSort.head(7)
hbBooks

,rownames,volume,area,weight,cover
0,1,885,382,800,hb
1,2,1016,468,950,hb
2,3,1125,387,1050,hb
3,4,239,371,350,hb
4,5,701,371,750,hb
5,6,641,367,600,hb
6,7,1228,396,1075,hb


Cleaning Log V

In [5]:
print(f'The mean weight of all hardback books is {hbBooks['weight'].mean()} grams.')

The mean weight of all hardback books is 796.4285714285714 grams.


Cleaning Log VI

verify the mean weight of hardback books by hand:
=====================================================
The mean is 796.4285714285714 -> (800+950+1050+350+750+600+1075)/7 = 796.4285714285714

Cleaning Log VII

In [6]:
pbBooks = coverSort.tail(8)
pbBooks

,rownames,volume,area,weight,cover
7,8,412,0,250,pb
8,9,953,0,700,pb
9,10,929,0,650,pb
10,11,1492,0,975,pb
11,12,419,0,350,pb
12,13,1010,0,950,pb
13,14,595,0,425,pb
14,15,1034,0,725,pb


Cleaning Log VIII

In [7]:
print(f'The mean weight of all paperback books is {pbBooks['weight'].mean()} grams.')

The mean weight of all paperback books is 628.125 grams.


Cleaning Log IX

In [8]:
print(f'The median weight of all books is {df['weight'].median()} grams.')
print(f'The median weight of all hardback books is {hbBooks['weight'].median()} grams.')
print(f'The median weight of all paperback is {pbBooks['weight'].median()} grams.')

The median weight of all books is 725.0 grams.
The median weight of all hardback books is 800.0 grams.
The median weight of all paperback is 675.0 grams.


Cleaning Log X

Data Analysis Conclusions
=============================================================================================================
1. The mean weight of all books (~706.67 g) is close to the median weight of all books, therefore median weight of all books has a normal distribution.
2. The mean weight of all hardback books (~796.4e g). When considering all books, most of the weight is with the hardback books.
3. The mean weight of all paperback books (~628.13 g) is close to the median weight of all paperback books, therefore median weight of paperback also has a normal distribution.

URL for This Assignmemnt's Repo
================================================================================

https://github.com/chillyvee70/cs82a-portfolio/tree/main/module4